core/services/ocr2/plugins/ocr2keeper/evm21/block_subscriber.go

```diff
@@ -56,6 +57,10 @@ type BlockSubscriber struct {
 	lggr             logger.Logger
 }
 
+func (bs *BlockSubscriber) LatestBlock() *ocr2keepers.BlockKey {
+	return bs.latestBlock.Load()
+}
+
```

Comment:
i'm not sure why this change was needed. was this moved from somewhere?

Reply:
This blockSubscriber is used in StreamsLookup, to be more precise, StreamsLookup uses only the latestBlock from blockSubscriber. This blockSubscriber is heavy and from a heavy package. we dont want to carry this to the new streams lib.

So, in this refactoring, `streams` takes a latestBlockProvider interface and requires implementation of LatestBlock function. Check streams.go L#71 

Would like to hear if there is better way to achieve this.

Reply:
sounds good